pkg/manifestcomparators/comp_must_not_exceed_cost_budget.go

```diff
@@ -184,3 +208,101 @@ func extractCELContext(schemas []*apiextensionsv1.JSONSchemaProps) (*apiextensio
 
 	return celContext, nil
 }
+
+// getUnboundedParentFields returns a list of field paths that have unbounded cardinality in the ancestry path.
+// This is aiming to help users identify where the unbounded cardinality is coming from.
+func getUnboundedParentFields(ancestry []*apiextensionsv1.JSONSchemaProps, fldPath *field.Path) []string {
+	cleanPathParts := getCleanPathParts(fldPath)
+	var path *field.Path
+
+	unboundedParents := []string{}
+	for i, s := range ancestry {
```

Comment:
let's check that the `cleanPathParts` len equals to the len of `ancestry` before we run the loop